daemon/daemon.go

```diff
@@ -973,6 +974,20 @@ func (daemon *Daemon) setHostConfig(container *Container, hostConfig *runconfig.
 
 	container.Lock()
 	defer container.Unlock()
+
+	var stat syscall.Stat_t
```

Comment:
we cannot call `syscall.Stat_t` and `syscall.Stat` like this, it's platform dependent. Can you extract this code to a function in `daemon/daemon_unix.go` and `daemon/daemon_windows.go`?